server/enhancedDate.js

```diff
@@ -1 +1,72 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+var _date;
+var weekday = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+var month = ["January", "February", "March", "April", "May", "June", "July", "Auguest", "September", "October", "Novenber", "December"];
+var current = new Date();
+var dateIsSet = false;
+
+function _setDate(oDate) {
+	if(oDate instanceof Date) {
+		_date = oDate;
+	} else {
```

Comment:
This function should also set the date based on a millisecond value passed in if a number has been passed in.